x-parser.js

```diff
@@ -239,61 +237,57 @@ export class XParser {
-      case '#128': return 'Invalid tag attribute interpolation (must use kebab-case names and double-quoted values).';
-      case '#129': return 'Invalid tag property interpolation (must use kebab-case names and double-quoted values).';
-      case '#130': return 'Invalid closing quote on tag attribute or property.';
+      case '#125': return 'Invalid tag attribute interpolation (must use kebab-case names and double-quoted values).';
+      case '#126': return 'Invalid tag property interpolation (must use kebab-case names and double-quoted values).';
+      case '#127': return 'Invalid closing quote on tag attribute or property.';
 
       case '#140': return 'CDATA sections are not supported. Use character references instead: https://developer.mozilla.org/en-US/docs/Glossary/Character_reference.';
 
       case '#150': return 'Bad escape in tagged template string. Use “&bsol;” for “\\”, “&dollar;” for “\\$”, and “&grave;” for “\\`”. All character references are supported: https://developer.mozilla.org/en-US/docs/Glossary/Character_reference.';
       case '#151': return 'Ambiguous ampersand character or invalid hexadecimal character reference.';
       case '#152': return 'Invalid comment. Comments cannot start with “>” or “->” characters, they cannot include a set of “--” characters, and they cannot end with a “-” character.';
       case '#153': return 'Unsupported native tag - supported native tags are listed here: https://github.com/Netflix/x-element/blob/main/doc/TEMPLATES.md#supported-native-tags.';
-      case '#154':
-      case '#155': return 'Invalid closing tag (all non-void start tags much have matching end tags).';
-      case '#156': return 'Unsupported <textarea> interpolation. Interpolation must be exact (<textarea>${…}</textarea>).';
-      case '#157': return 'Unsupported declarative shadow root on <template>. The “shadowrootmode” attribute is not supported.';
+      case '#154': return 'Invalid end tag (all non-void start tags much have matching end tags).';
+      case '#155': return 'Unsupported <textarea> interpolation. Interpolation must be exact (<textarea>${…}</textarea>).';
+      case '#156': return 'Unsupported declarative shadow root on <template>. The “shadowrootmode” attribute is not supported.';
+      case '#157': return 'Missing closing quote on bound attribute or property.';
     }
   }
 
   // Block #100-#119 — Invalid transition errors.
   static #getErrorMessageKeyFromValue(value) {
+    // Note that the following states _never_ have their own generic errors. The
+    //  main reason is because they all encode a lookahead which is a subset of
+    //  a related “*Malformed” pattern.
+    // - XParser.#startTagOpen  (caught by XParser.#startTagSpaceMalformed)
+    // - XParser.#boolean       (caught by XParser.#startTagSpaceMalformed)
+    // - XParser.#attribute     (caught by XParser.#startTagSpaceMalformed)
+    // - XParser.#danglingQuote (caught by XParser.#startTagSpaceMalformed)
```

Comment:
Added commentary as to why the list below doesn’t include _all_ the `value` states. Kinda neat that we were able to get our code coverage tool to find unreachable cases for us 🎉